### PR TITLE
updated helm values to use workoad identity 

### PR DESCRIPTION
**Pull request by NickAzureDevops**
### Jira link (https://tools.hmcts.net/jira/browse/DTSPO-14630)

updated helm values to use workoad identity 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

## 🤖AEP PR SUMMARY🤖

_I'm a bot that generates AI summaries of pull requests, see [AEP](https://kainossoftwareltd.github.io/ai-enhanced-platform/) for more details_

### File `apps/monitoring/helm-base-versions/helm-base-versions.yaml`
- Removed the `labels` field under `metadata`.
- Added the label `azure.workload.identity/use: \"true\"` under `template.metadata.labels`.
- Updated the value of `API_SERVER_URL` to \"https://kubernetes.default.svc.cluster.local\".

### CODE REVIEW

**Review by thomast1906 — CHANGES_REQUESTED** [review]
Is that PR correct with moving the metadata / labels position?

**Comment by NickAzureDevops** [comment]
> Is that PR correct with moving the metadata / labels position?

i think so as same as this here 

https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster that reference this

`cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: sample-workload-identity
  namespace: ${SERVICE_ACCOUNT_NAMESPACE}
  labels:
    azure.workload.identity/use: "true"  # Required. Only pods with this label can use workload identity.
spec:
  serviceAccountName: ${SERVICE_ACCOUNT_NAME}
  containers:
    - image: <image>
      name: <containerName>
EOF`

File: apps/monitoring/serviceaccount/aat.yaml [hunk 1]
```diff
@@ -4,4 +4,5 @@ metadata:
   name: ${NAMESPACE}
   namespace: ${NAMESPACE}
   annotations:
-    azure.workload.identity/client-id: "fe91170e-e2a5-421a-90d0-d7fefab8382c"
\ No newline at end of file
+    azure.workload.identity/client-id: "fe91170e-e2a5-421a-90d0-d7fefab8382c"
+    azure.workload.identity/use: "true"
```
Comment (MartyFox) [inline apps/monitoring/serviceaccount/aat.yaml:8]: Dont think you need this here.
[Chart library](https://github.com/search?q=repo%3Ahmcts%2Fchart-library%20azure.workload.identity%2Fuse&type=code) shows that annotation/label is only use on the pod template

**Comment by NickAzureDevops** [comment]
added to all env except ptl,ptlsbox and sbox that have the cronjob there

**Comment by NickAzureDevops** [comment]
> added to all env except ptl,ptlsbox and sbox that have the cronjob there

this is no longer required

**Review by MartyFox — APPROVED** [review]

**Pull request merged**